src/main/java/com/synopsys/integration/detect/configuration/DetectConfigurationFactory.java

```diff
@@ -196,8 +197,8 @@ public PhoneHomeOptions createPhoneHomeOptions() {
     public DetectToolFilter createToolFilter(RunDecision runDecision, BlackDuckDecision blackDuckDecision) {
         Optional<Boolean> impactEnabled = Optional.of(detectConfiguration.getValue(DetectProperties.DETECT_IMPACT_ANALYSIS_ENABLED.getProperty()));
 
-        FilterableEnumList<DetectTool> includedTools = getValue(DetectProperties.DETECT_TOOLS);
-        FilterableEnumList<DetectTool> excludedTools = getValue(DetectProperties.DETECT_TOOLS_EXCLUDED);
+        AllNoneEnumCollection<DetectTool> includedTools = PropertyConfigUtils.getAllNoneList(detectConfiguration, DetectProperties.DETECT_TOOLS.getProperty());
```

Comment:
It doesn't give me the warm and fuzzies. I would be for local wrapping methods in each factory

Reply:
Yeah I agree, it' an artifact of the hierarchy and reusing the ExtendedEnumList. I considered not subclassing but then I need to reimplement both ExtendedEnumProperty and ListProperty. I like that I was able to basically reuse all of them. I think we could improve ListProperty to let you return a specify the List subclass (maybe through generics) but that seemed like going to far for this PR.

Reply:
Can we make a ticket for this?

Reply:
Done IDETECT-3055.